src/components/ContentNode.vue

```diff
@@ -468,14 +468,16 @@ export default {
     // Dynamically map each content item and any children to their
     // corresponding components, and wrap the whole tree in a <div>
     return createElement(this.tag, { class: 'content' }, (
-      this.content.map(renderNode(createElement, this.references), this)
+      this.content.map(renderNode(createElement, this.store.state.references), this)
     ));
   },
   inject: {
-    references: {
-      default() {
-        return {};
-      },
+    store: {
+      default: () => ({
+        state: {
+          references: {},
+        },
+      }),
     },
```

Comment:
why in some components we are injecting default values and in some others we don't?

```js
inject: {
    store: {
      default: () => ({
        state: {
          references: {},
        },
      }),
    },
 },
```

vs

```js
inject: ['store'],
```
do we need these default values at all?

Reply:
I'm curious about that as well. I think in certain components we can be sure that we always provided the reference, but that's not true in other components. @mportiz08 Do you know why we sometimes don't provide defaults when injecting?

Reply:
Mostly we don't ever need defaults since some top-level component is always providing the things we're depending on.

Defaults are super helpful for scenarios where you want to make the component more flexible to handle cases where those things aren't provided. This shows up a lot in unit tests specifically where we usually shallow mount things independently from the parent component that would usually provide these things 😅